ord_schema/validations.py

```diff
@@ -0,0 +1,428 @@
+            schema.CompoundIdentifier: ValidateCompoundIdentifier,
+            # Setup
+            schema.Vessel: ValidateVessel,
+            schema.ReactionSetup: ValidateReactionSetup,
+            # Conditions
+            schema.ReactionConditions: ValidateReactionConditions,
+            schema.TemperatureConditions: ValidateTemperatureConditions,
+            schema.TemperatureConditions.Measurement: ValidateTemperatureMeasurement,
+            schema.PressureConditions: ValidatePressureConditions,
+            schema.PressureConditions.Measurement: ValidatePressureMeasurement,
+            schema.StirringConditions: ValidateStirringConditions,
+            schema.IlluminationConditions: ValidateIlluminationConditions,
+            schema.ElectrochemistryConditions: ValidateElectrochemistryConditions,
+            schema.ElectrochemistryConditions.Measurement: ValidateElectrochemistryMeasurement,
+            schema.FlowConditions: ValidateFlowConditions,
+            schema.FlowConditions.Tubing: ValidateTubing,
+            # Annotations
+            schema.ReactionNotes: ValidateReactionNotes,
+            schema.ReactionObservation: ValidateReactionObservation,
+            # Outcome
+            schema.ReactionWorkup: ValidateReactionWorkup,
+            schema.ReactionOutcome: ValidateReactionOutcome,
+            schema.ReactionProduct: ValidateReactionProduct,
+            schema.Selectivity: ValidateSelectivity,
+            schema.DateTime: ValidateDateTime,
+            schema.ReactionAnalysis: ValidateReactionAnalysis,
+            # Metadata
+            schema.ReactionProvenance: ValidateReactionProvenance,
+            schema.Person: ValidatePerson,
+            # Units
+            schema.Time: ValidateTime,
+            schema.Mass: ValidateMass,
+            schema.Moles: ValidateMoles,
+            schema.Volume: ValidateVolume,
+            schema.Concentration: ValidateConcentration,
+            schema.Pressure: ValidatePressure,
+            schema.Temperature: ValidateTemperature,
+            schema.Current: ValidateCurrent,
+            schema.Voltage: ValidateVoltage,
+            schema.Length: ValidateLength,
+            schema.Wavelength: ValidateWavelength,
+            schema.FlowRate: ValidateFlowRate,
+            schema.Percentage: ValidatePercentage,
+        }[type(message)](message)
+    except KeyError:
+        # NOTE(ccoley): I made the conscious decision to raise an error here,
+        # rather than assume that the message is valid. If a message does not
+        # require any message-level checks (not uncommon), then it should still
+        # be listed in the dictionary switch above withpass. This will force
+        # us to think about what is necessary if/when new messages are added.
+        raise NotImplementedError(f"Don't know how to validate {type(message)}")
+
+
+class ValidationWarning(Warning):
+    pass
+
+def return_message_if_valid(func):
```

Comment:
This seems like an unnecessary redirection; why not just have each function return the message?

Reply:
I'd still like this change.

Reply:
It was more aesthetically pleasing to have the same wrapper preceding every function instead of having ```return message``` at the end of each one. But changing now